File: backend/submissions/task2_exam1_student6/submission.py
# Hàm kiểm tra số hoàn hảo
def is_perfect_number(x):
    if x < 2:
        return False
    sum_divisors = sum(i for i in range(1, x) if x % i == 0)
    return sum_divisors == x

# Đọc dữ liệu từ file input
with open("HOANHAO.INP", "r") as f:
    n = int(f.readline().strip())
    a = list(map(int, f.readline().split()))

# Tính tổng các số hoàn hảo
perfect_sum = sum(x for x in a if is_perfect_number(x))

# Ghi kết quả ra file output
with open("HOANHAO.OUT", "w") as f:
    f.write(str(perfect_sum if perfect_sum > 0 else -1) + "\n")

File: backend/submissions/task2_exam1_student6/submission.cpp
#include <iostream>
#include <fstream>
#include <vector>

using namespace std;

// Hàm kiểm tra số hoàn hảo
bool isPerfectNumber(int x) {
    if (x <= 1) return false;
    int sum = 1;
    for (int i = 2; i * i <= x; i++) { 
        if (x % i == 0) { 
            sum += i; 
            if (i != x / i) sum += x / i;
        }
    }
    return sum == x;
}

int main() {
    ifstream infile("HOANHAO.INP");
    ofstream outfile("HOANHAO.OUT");

    if (!infile || !outfile) return 1;

    vector<int> numbers;
    int num;
    while (infile >> num) {
        numbers.push_back(num);
    }

    infile.close();

    int sumPerfect = 0;
    for (int num : numbers) {
        if (isPerfectNumber(num)) {
            sumPerfect += num;
        }
    }

    // Chỉ ghi kết quả ra file, không ghi debug
    outfile << (sumPerfect > 0 ? sumPerfect : -1);
    outfile.close();

    return 0;
}
